packages/nx-aws-cache/src/tasks-runner/aws-cache.ts

```diff
@@ -202,11 +223,16 @@ export class AwsCache implements RemoteCache {
         Key: this.getS3Key(tgzFileName),
       });
 
+    // eslint-disable-next-line max-lines
     try {
       const commandOutput = await this.s3.send(params);
       const fileStream = commandOutput.Body as Readable;
-
-      await pipelinePromise(fileStream, writeFileToLocalDir);
+      if (this.encryptConfig) {
+        const decryptStream = new Decrypt(this.encryptConfig);
```

Comment:
The same as for Encrypt, make this a singleton.

Reply:
Singleton  will damage files

Reply:
What is the performance hit for this process, have you measured it?

Reply:
Why did you use createReadStream on each upload, have you measured it? 
Decrypt is stream transform like createReadStream how do you want to encrypt data with IV without state?